dpnp/dpnp_iface_manipulation.py

```diff
@@ -1469,7 +1469,20 @@ def copyto(dst, src, casting="same_kind", where=True):
             f"but got {type(dst)}"
         )
     if not dpnp.is_supported_array_type(src):
+        src_orig = src
         src = dpnp.array(src, sycl_queue=dst.sycl_queue)
+        if not hasattr(src_orig, "dtype"):
```

Comment:
It seems it's not needed to keep the reference on source object:
```suggestion
        no_dtype_attr = not hasattr(src_orig, "dtype")
        src = dpnp.array(src, sycl_queue=dst.sycl_queue)
        if no_dtype_attr:
```